internal/command/clistate/state.go

```diff
@@ -114,8 +114,13 @@ func (l *locker) Lock(s statemgr.Locker, reason string) tfdiags.Diagnostics {
 
 	l.state = s
 
-	ctx, cancel := context.WithTimeout(l.ctx, l.timeout)
-	defer cancel()
+	// only assign a context with timeout if the locker timeout is not 0
+	ctx := l.ctx
+	if l.timeout != 0 {
+		c, cancel := context.WithTimeout(l.ctx, l.timeout)
+		defer cancel()
+		ctx = c
+	}
```

Comment:
**(nitpick/suggestion)** It may be an edge case, but technically the value of `l.timeout` could be negative, so I'd like to propose the adjustment to cover that case.

```suggestion
	if l.timeout > 0 {
		ctx, cancel := context.WithTimeout(l.ctx, l.timeout)
		defer cancel()
	}
```